graphrag_sdk/models/litellm.py

```diff
@@ -0,0 +1,283 @@
+import os
+import logging
+from ollama import Client
+from typing import Optional
+from litellm import completion
+from .model import (
+    OutputMethod,
+    GenerativeModel,
+    GenerativeModelConfig,
+    GenerationResponse,
+    FinishReason,
+    GenerativeModelChatSession,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO) 
+
+class LiteModel(GenerativeModel):
+    """
+    A generative model that interfaces with the LiteLLM for chat completions.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        generation_config: Optional[GenerativeModelConfig] = None,
+        system_instruction: Optional[str] = None,
+    ):
+        """
+        Initialize the LiteModel with the required parameters.
+
+        Args:
+            model_name (str): The name and the provider for the LiteLLM client.
+            generation_config (Optional[GenerativeModelConfig]): Configuration settings for generation.
+            system_instruction (Optional[str]): Instruction to guide the model.
+        """
+
+        self.model = model_name
+        if "/" in model_name:
+            self.provider = model_name.split("/")[0]
+            self.model_name = model_name.split("/")[1]
+        else:
+            self.provider = "openai"
+            self.model_name = model_name
+        
+        self.credentials_validation()
+        self.generation_config = generation_config or GenerativeModelConfig()
+        self.system_instruction = system_instruction
+        
+    def credentials_validation(self) -> None:
+        """
+        Validate the credentials for the model.
+
+        Returns:
+            bool: True if the credentials are valid, False otherwise.
+        """
+        if self.provider == "openai":
+            if not os.getenv("OPENAI_API_KEY"):
+                raise ValueError("Missing OpenAI API key in the environment.")
+               
+        elif self.provider == "azure":
+            if not os.getenv("AZURE_API_KEY") or not os.getenv("AZURE_API_BASE") or not os.getenv("AZURE_API_VERSION"):
+                raise ValueError("Missing Azure-OpenAI credentials in the environment. See the Github Readme for more information.")
+        
+        elif self.provider == "gemini":
+            if not os.getenv("GEMINI_API_KEY"):
+                raise ValueError("Missing Gemini API key in the environment.")
+        
+        elif self.provider == "ollama":
+            self.ollama_client = Client()
+            self.check_and_pull_model()
```

Comment:
are these all of the different LLM vendor litellm supports ?

Reply:
There are more, but I noticed that they perform credentials validation poorly